server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/IMAPServerTest.java

```diff
@@ -1211,6 +1219,72 @@ void oauthShouldFailWhenIntrospectTokenServerError() throws Exception {
             client.sendCommand("AUTHENTICATE OAUTHBEARER " + oauthBearer);
             assertThat(client.getReplyString()).contains("NO AUTHENTICATE processing failed.");
         }
+
+        @Test
+        void oauthShouldImpersonateFailWhenNOTDelegated() throws Exception {
+            String oauthBearer = OIDCSASLHelper.generateOauthBearer(USER2.asString(), OidcTokenFixture.VALID_TOKEN);
+            IMAPSClient client = imapsClient(port);
+            client.sendCommand("AUTHENTICATE OAUTHBEARER " + oauthBearer);
+            assertThat(client.getReplyString()).contains("NO AUTHENTICATE Delegation is forbidden");
+        }
+
+        @Test
+        void oauthShouldImpersonateSuccessWhenDelegated() throws Exception {
+            createImapServerForImpersonateTest();
+
+            String oauthBearer = OIDCSASLHelper.generateOauthBearer(USER2.asString(), OidcTokenFixture.VALID_TOKEN);
+            IMAPSClient client = imapsClient(port);
+            client.sendCommand("AUTHENTICATE OAUTHBEARER " + oauthBearer);
+            assertThat(client.getReplyString()).contains("OK AUTHENTICATE completed.");
+        }
+
+        @Test
+        void impersonationShouldWorkWhenDelegated() throws Exception {
+            createImapServerForImpersonateTest();
+
+            // USER2: append a message
+            try (TestIMAPClient client = new TestIMAPClient(imapsClient(port))) {
+                client.login(USER2.asString(), USER_PASS)
+                    .append("INBOX", SMALL_MESSAGE);
+            }
+
+            // USER1 authenticate and impersonate as USER2
+            try (TestIMAPClient client = new TestIMAPClient(imapsClient(port))) {
+                String oauthBearer = OIDCSASLHelper.generateOauthBearer(USER2.asString(), OidcTokenFixture.VALID_TOKEN);
+                String authenticateResponse = client.sendCommand("AUTHENTICATE OAUTHBEARER " + oauthBearer);
+                assertThat(authenticateResponse).contains("OK AUTHENTICATE completed.");
+
+                assertThat(client.select("INBOX")
+                    .readFirstMessage()).contains(SMALL_MESSAGE);
+            }
+        }
+
+        private void createImapServerForImpersonateTest() throws Exception {
```

Comment:
Suggestion : always set this up in @BeforeEach.

For tests failing delegation you can use USER3 which USER have no access to.